db_service/migrations.go

```diff
@@ -33,114 +33,43 @@ func RunMigrations(db *gorm.DB) error {
 
 	// initial migration - creates tables
 	migrations[0] = func() error {
-		if err := db.Exec(`CREATE TABLE IF NOT EXISTS service_instance_details (
-			  id varchar(255) NOT NULL DEFAULT '',
-			  created_at timestamp NULL DEFAULT NULL,
-			  updated_at timestamp NULL DEFAULT NULL,
-			  deleted_at timestamp NULL DEFAULT NULL,
-			  name varchar(255) DEFAULT NULL,
-			  location varchar(255) DEFAULT NULL,
-			  url varchar(255) DEFAULT NULL,
-			  other_details text,
-			  service_id varchar(255) DEFAULT NULL,
-			  plan_id varchar(255) DEFAULT NULL,
-			  space_guid varchar(255) DEFAULT NULL,
-			  organization_guid varchar(255) DEFAULT NULL,
-			  PRIMARY KEY (id)
-			) ENGINE=InnoDB DEFAULT CHARSET=utf8`).Error; err != nil {
-			return err
-		}
-		if err := db.Exec(`CREATE TABLE IF NOT EXISTS service_binding_credentials (
-			  id int(10) unsigned NOT NULL AUTO_INCREMENT,
-			  created_at timestamp NULL DEFAULT NULL,
-			  updated_at timestamp NULL DEFAULT NULL,
-			  deleted_at timestamp NULL DEFAULT NULL,
-			  other_details text,
-			  service_id varchar(255) DEFAULT NULL,
-			  service_instance_id varchar(255) DEFAULT NULL,
-			  binding_id varchar(255) DEFAULT NULL,
-			  PRIMARY KEY (id),
-			  KEY idx_service_binding_credentials_deleted_at (deleted_at)
-			) ENGINE=InnoDB DEFAULT CHARSET=utf8`).Error; err != nil {
-			return err
-		}
-		if err := db.Exec(`CREATE TABLE IF NOT EXISTS provision_request_details (
-			  id int(10) unsigned NOT NULL AUTO_INCREMENT,
-			  created_at timestamp NULL DEFAULT NULL,
-			  updated_at timestamp NULL DEFAULT NULL,
-			  deleted_at timestamp NULL DEFAULT NULL,
-			  service_instance_id varchar(255) DEFAULT NULL,
-			  request_details varchar(255) DEFAULT NULL,
-			  PRIMARY KEY (id),
-			  KEY idx_provision_request_details_deleted_at (deleted_at)
-			) ENGINE=InnoDB DEFAULT CHARSET=utf8`).Error; err != nil {
-			return err
-		}
-		if err := db.Exec(`CREATE TABLE IF NOT EXISTS plan_details (
-			  id varchar(255) NOT NULL DEFAULT '',
-			  created_at timestamp NULL DEFAULT NULL,
-			  updated_at timestamp NULL DEFAULT NULL,
-			  deleted_at timestamp NULL DEFAULT NULL,
-			  service_id varchar(255) DEFAULT NULL,
-			  name varchar(255) DEFAULT NULL,
-			  features text,
-			  PRIMARY KEY (id)
-			) ENGINE=InnoDB DEFAULT CHARSET=utf8`).Error; err != nil {
-			return err
-		}
-		if err := db.Exec(`CREATE TABLE IF NOT EXISTS migrations (
-			  id int(10) unsigned NOT NULL AUTO_INCREMENT,
-			  created_at timestamp NULL DEFAULT NULL,
-			  updated_at timestamp NULL DEFAULT NULL,
-			  deleted_at timestamp NULL DEFAULT NULL,
-			  migration_id int(10) DEFAULT NULL,
-			  PRIMARY KEY (id)
-			) ENGINE=InnoDB DEFAULT CHARSET=utf8`).Error; err != nil {
-			return err
-		}
-		return nil
+		return autoMigrateTables(db,
+			&models.ServiceInstanceDetailsV1{},
+			&models.ServiceBindingCredentialsV1{},
+			&models.ProvisionRequestDetailsV1{},
+			&models.PlanDetailsV1{},
+			&models.MigrationV1{})
```

Comment:
Results for: `show create table migrations`

Note: the AUTO_INCREMENT index starts at 4 because the query was run _after_ migrations had run and 3 entries have been made.

```
CREATE TABLE `migrations` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `created_at` timestamp NULL DEFAULT NULL,
  `updated_at` timestamp NULL DEFAULT NULL,
  `deleted_at` timestamp NULL DEFAULT NULL,
  `migration_id` int(10) DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `idx_migrations_deleted_at` (`deleted_at`)
) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=utf8
```